src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -0,0 +1,513 @@
+            var firstLine = changedText.Lines[(int)changedBodyRange.Start.Line];
+            var desiredIndentationLevel = context.Indentations[firstLine.LineNumber].IndentationLevel;
+            var desiredIndentation = GetIndentationString(context, desiredIndentationLevel);
+            var firstNonWhitespaceOffset = firstLine.GetFirstNonWhitespaceOffset();
+            if (firstNonWhitespaceOffset.HasValue)
+            {
+                var edit = new TextEdit()
+                {
+                    Range = new Range(
+                        new Position(firstLine.LineNumber, 0),
+                        new Position(firstLine.LineNumber, firstNonWhitespaceOffset.Value)),
+                    NewText = desiredIndentation
+                };
+                edits.Add(edit);
+            }
+
+            // We should also move any code that comes after '{' down to its own line.
+            var originalInnerCodeBlockSpan = TextSpan.FromBounds(innerCodeBlock.Position, innerCodeBlock.EndPosition);
+            TrackChangeInSpan(sourceText, originalInnerCodeBlockSpan, changedText, out var changedInnerCodeBlockSpan, out _);
+            var innerCodeBlockRange = changedInnerCodeBlockSpan.AsRange(changedText);
+
+            var innerCodeBlockLine = changedText.Lines[(int)innerCodeBlockRange.Start.Line];
+            var textAfterBlockStart = innerCodeBlockLine.ToString().Substring(innerCodeBlock.Position - innerCodeBlockLine.Start);
+            var isBlockStartOnSeparateLine = string.IsNullOrWhiteSpace(textAfterBlockStart);
+            var innerCodeBlockIndentationLevel = desiredIndentationLevel + 1;
+            var desiredInnerCodeBlockIndentation = GetIndentationString(context, innerCodeBlockIndentationLevel);
+            var whitespaceAfterBlockStart = textAfterBlockStart.GetLeadingWhitespace();
+
+            if (!isBlockStartOnSeparateLine)
+            {
+                // If the first line contains code, add a newline at the beginning and indent it.
+                var edit = new TextEdit()
+                {
+                    Range = new Range(
+                        new Position(innerCodeBlockLine.LineNumber, innerCodeBlock.Position - innerCodeBlockLine.Start),
+                        new Position(innerCodeBlockLine.LineNumber, innerCodeBlock.Position + whitespaceAfterBlockStart.Length - innerCodeBlockLine.Start)),
+                    NewText = Environment.NewLine + desiredInnerCodeBlockIndentation
+                };
+                edits.Add(edit);
+            }
+            else
+            {
+                // The code inside the code block directive is on its own line. Ideally the C# formatter would have already taken care of it.
+                // Except, the first line of the code block is not indented because of how our SourceMappings work.
+                // The edit for the first line is filtered out due to lack of a matching source mapping.
```

Comment:
woa really? How does that work in VS? Aka I imagine in VS you wouldn't have proper C# coloring/mapping going on.

Reply:
I think we already spoke about this behavior offline.  When the edit being returned start from before the open curly, we will completely ignore it even though that edit when combined with other edits would produce the correct output.